File: js/test/test_get_status.js
const WebSocket = require('ws');

// 1. 配置
const CONFIG = {
    url: 'ws://192.168.31.233:3000?access_token=test',
    action: 'get_status',
    echo: 'status_check_001'
};

const ws = new WebSocket(CONFIG.url);

// 2. 监听连接打开
ws.on('open', () => {
    console.log(`✅ 已建立 WebSocket 连接: ${CONFIG.url}`);
    
    // 构造 OneBot 标准请求包
    const payload = {
        action: CONFIG.action,
        params: {},
        echo: CONFIG.echo
    };

    console.log(`🚀 正在请求接口: ${CONFIG.action}...`);
    ws.send(JSON.stringify(payload));
});

// 3. 监听消息返回
ws.on('message', (rawData) => {
    try {
        const response = JSON.parse(rawData.toString());

        // 排除掉 lifecycle 等 meta_event，只处理我们发送的那个请求的响应
        if (response.echo === CONFIG.echo) {
            console.log('\n--- 收到状态响应 ---');
            console.log(JSON.stringify(response, null, 2));

            if (response.status === 'ok') {
                const { online, good } = response.data;

                console.log('\n========== 状态检测结果 ==========');
                console.log(`在线状态 (online): ${online ? '🟢 在线' : '🔴 离线'}`);
                console.log(`运行正常 (good):   ${good ? '✅ 是' : '⚠️ 否'}`);

                if (online && good) {
                    console.log('\n✨ Bot 运行完美！');
                } else {
                    console.log('\n🚨 请检查 Bot 运行环境或登录状态！');
                }
            }
            
            // 查完就走，关闭连接
            ws.close();
        } else if (response.post_type === 'meta_event') {
            // 这里可以打印心跳或连接信息，不想看可以注掉
            console.log(`[系统事件] ${response.meta_event_type}: ${response.sub_type || ''}`);
        }
    } catch (err) {
        console.error('解析消息失败:', err);
    }
});

// 4. 异常处理
ws.on('error', (err) => {
    if (err.message.includes('ECONNREFUSED')) {
        console.error('❌ 连接被拒绝：NapCat 服务可能没启动，或者 IP/端口填错了。');
    } else {
        console.error('❌ 发生错误:', err.message);
    }
});

ws.on('close', () => {
    console.log('\n🔌 连接已断开');
});

File: py/test/test_get_status.py
import asyncio
import websockets
import json

async def get_bot_status():
    # 你的服务器地址和 Token
    uri = "ws://192.168.31.233:3000?access_token=test"
    
    try:
        # 1. 建立连接
        async with websockets.connect(uri) as websocket:
            print(f"已连接到: {uri}")

            # 2. 接收连接成功的第一个生命周期消息 (lifecycle)
            init_msg = await websocket.recv()
            print(f"收到初始信号: {init_msg}")

            # 3. 构造请求数据 (Action)
            payload = {
                "action": "get_status",
                "params": {},
                "echo": "123"
            }

            # 4. 发送请求
            print(f"正在发送 Action: {payload['action']}...")
            await websocket.send(json.dumps(payload))

            # 5. 等待并接收响应
            response = await websocket.recv()
            result = json.loads(response)

            # 6. 解析并美化输出
            print("\n--- 获取成功 ---")
            print(json.dumps(result, indent=4, ensure_ascii=False))
            
            if result.get("status") == "ok":
                data = result.get("data", {})
                online = data.get("online")
                good = data.get("good")
                
                print(f"\n========== 状态检测结果 ==========")
                print(f"在线状态 (online): {online}")
                print(f"运行正常 (good):   {good}")
                
                if online:
                    print("\n✅ Bot 当前在线！")
                else:
                    print("\n❌ Bot 已离线！可能被顶下线了")
                    
                if not good:
                    print("⚠️  Bot 状态异常！")

    except websockets.exceptions.ConnectionClosedError as e:
        print(f"连接被关闭: {e}")
        print("❌ 可能 Bot 已离线，无法建立连接")
    except ConnectionRefusedError:
        print("❌ 连接被拒绝，NapCat 服务可能未运行")
    except Exception as e:
        print(f"连接或通信发生错误: {e}")

if __name__ == "__main__":
    asyncio.run(get_bot_status())
